cumulus_library_opioid/vocab/static_builder.py

```diff
@@ -0,0 +1,131 @@
+import csv
+import dataclasses
+import pathlib
+
+import pandas
+
+from cumulus_library import base_table_builder, base_utils, study_manifest
+from cumulus_library.template_sql import base_templates
+
+@dataclasses.dataclass
+class TableConfig:
+    file_path: str
+    delimiter: str
+    table_name: str
+    headers: list[str]
+    dtypes: dict
+    parquet_types: list[str]
+    filtered_name: str | None = None
+    ignore_header: bool = False
+    map_cols: list[dict] | None = None
+
+class StaticBuilder(base_table_builder.BaseTableBuilder):
+    display_text = "Building static data tables..."
+    base_path = pathlib.Path(__file__).resolve().parent
+
+    tables = [
+        TableConfig(
+            base_path / './common/keywords/keywords.curated.tsv',
+            '\t',
+            'keywords',
+            ['STR'],
+            {"STR": 'str'},
+            ['STRING'],
+            filtered_name= base_path / './common/keywords/keywords.curated.filtered.tsv',
+        ),
+        TableConfig(
+            base_path / './common/expand_rules/expand_rules.tsv',
+            '\t',
+            'expand_rules',
+            # TODO: do we actually need to preserve 'rule'?
+            ["TTY1","RELA","TTY2","rule", ],
+            {"TTY1":'str',"RELA":'str',"TTY2":'str',"rule":'str'},
+            ['STR','STR','STR','STR','STR'],
+            ignore_header=True,
+            map_cols = [{'from':'rule','to':'include',"map_dict":{ 'yes': True, 'no':False }}]
+        )
+    ]
+
+    def filter_duplicated(self, path, delimiter, filtered_name):
+        target = path.with_suffix(".filtered.tsv")
+        #if target.exists():
+        #    return
+        with open(path) as file:
```

Comment:
Linting/ruff on this repo would probably find some good stuff, like the missing `encoding=` arg here. I know we control the source data, so it's not technically an issue here. But still, always good to document the encoding.

Reply:
Ok, ruff config is in, did a pass, ignored some errors on files that are likely to be replaced later.